### PR TITLE
Detects possible errors in user calculations in the admin and alerts …

### PR DESCRIPTION
**Pull request by LauraSteiner**
…user.

### CODE REVIEW

File: js/formidable_admin.js [hunk 1]
```diff
@@ -607,6 +607,140 @@ function frmAdminBuildJS(){
...
+	 * Checks a calculation for shortcodes that shouldn't be in it and returns a message if found.
+	 * @param calculation
+	 * @param inputElement
+	 * @returns {string}
+	 */
+	function checkShortcodes( calculation, inputElement ) {
+		var nonNumericShortcodesMessage = checkNonNumericShortcodes( calculation, inputElement );
+		var nonFormShortcodesMessage = checkNonFormShortcodes( calculation );
+
+		var msg = nonNumericShortcodesMessage;
```
Comment (stephywells) [inline js/formidable_admin.js]: This could be simplified a bit. How about 
```
var msg = checkNonNumericShortcodes( calculation, inputElement );
msg += checkNonFormShortcodes( calculation );
```

File: js/formidable_admin.js [hunk 2]
```diff
@@ -607,6 +607,140 @@ function frmAdminBuildJS(){
...
+		msg += nonFormShortcodesMessage;
+
+		return msg;
+	}
+
+	function checkCalculationCreatedByUser() {
```
Comment (stephywells) [inline js/formidable_admin.js]: Would you mind reordering the functions a bit so we can read them top to bottom like you did with your math shortcode? I find it hard to follow when I have to scroll up to find the functions.

File: js/formidable_admin.js (hunk 1 above)
Comment (LauraSteiner) [reply js/formidable_admin.js]: Nice!  Done.

File: js/formidable_admin.js (hunk 2 above)
Comment (LauraSteiner) [reply js/formidable_admin.js]: Happy to!  Done.

**Comment by LauraSteiner** [comment]
Thanks, Steph!  I appreciate your insights.

File: classes/controllers/FrmFormsController.php [hunk 3]
```diff
@@ -845,6 +845,10 @@ public static function form_classes( $form ) {
 		if ( isset( $form->options['js_validate'] ) && $form->options['js_validate'] ) {
 			echo ' frm_js_validate ';
 		}
+
+		if ( current_user_can( 'activate_plugins' ) && current_user_can( 'frm_edit_forms' ) ) {
```
Comment (stephywells) [inline classes/controllers/FrmFormsController.php]: Sorry... One more change. It looks like this should be in Pro since it isn't used here.

**Comment by LauraSteiner** [comment]
frm-admin-viewing has been removed from free and added to pro.
